selfdrive/car/hyundai/carcontroller.py

```diff
@@ -0,0 +1,174 @@
+from common.numpy_fast import clip, interp
+from selfdrive.boardd.boardd import can_list_to_can_capnp
+from selfdrive.car.hyundai.hyundaican import make_can_msg, create_lkas11, create_lkas12b
+from selfdrive.car.hyundai.values import ECU, STATIC_MSGS, CAR
+from selfdrive.can.packer import CANPacker
+
+
+# Steer torque limits
+STEER_MAX = 200   # Actual limit is about 1023, but not tested, and not needed
+STEER_DELTA = 5   # We have no Panda Safety, don't be silly here!   Good idea, YOU add Panda Safety!
+
+
+TARGET_IDS = [0x340]
+
+
+class CarController(object):
+  def __init__(self, dbc_name, car_fingerprint, enable_camera):
+    self.braking = False
+    self.controls_allowed = True
+    self.last_steer = 0
+    self.car_fingerprint = car_fingerprint
+    self.angle_control = False
+    self.idx = 0
+    self.lkas_request = 0
+    self.lanes = 0
+    self.steer_angle_enabled = False
+    self.ipas_reset_counter = 0
+    self.turning_inhibit = 0
+    self.limited_steer = 0
+    self.hide_lkas_fault = 100
+    print self.car_fingerprint
+
+    self.fake_ecus = set()
+    if enable_camera: self.fake_ecus.add(ECU.CAM)
+    self.packer = CANPacker(dbc_name)
+
+  def update(self, sendcan, enabled, CS, frame, actuators, CamS):
+    # When the driver starts inputting torque, reduce the requested torque
+    #  We do not want to remove torque because we want to know if OP is still steering
+    #  And how we do it, meh, who cares, this might do.
+    if abs(CS.steer_torque_driver) > 1:
+      self.limited_steer = actuators.steer / (abs(CS.steer_torque_driver) * 2)
```

Comment:
mmm... this causes a torque jump as soon as `abs(CS.steer_torque_driver) > 1` is true.

Maybe you can have a look how this is done for GM. You can see how the driver torque is taken into account to limit the commanded torque.

Reply:
GM looks nice, I will implement that.

This never used to have * 2, so there was no jump, but later... hack..